server/store.go

```diff
@@ -378,3 +385,34 @@ func (p *Plugin) getLastReminderTimeForUser(userID string) (int64, error) {
 
 	return reminderAt, nil
 }
+
+func (p *Plugin) saveReminderPreference(userID string, preference bool) error {
+	preferenceString := strconv.FormatBool(preference)
+	appErr := p.API.KVSet(reminderEnabledKey(userID), []byte(preferenceString))
+	if appErr != nil {
+		return errors.New(appErr.Error())
+	}
+	return nil
+}
+
+// getReminderPreference - gets user preference on reminder - default value will be true if in case any error
+func (p *Plugin) getReminderPreference(userID string) bool {
+	preferenceByte, appErr := p.API.KVGet(reminderEnabledKey(userID))
+	if appErr != nil {
+		log.Printf("error getting the reminder preference, err=%+v", appErr)
```

Comment:
Sorry, my previous comment was unclear. The plugin API actually exposes methods for logging. They should be used instead of `log`. You can find an example at https://github.com/openmohan/mattermost-plugin-todo/blob/24e1c44e136db59f6b4044c64ab21bcc36aa88a5/server/plugin.go#L117.

Reply:
> looks good in general, please add a test

Added test cases for `command.go`->`runSettingsCommand`.